src/cpp/fastdds/publisher/DataWriterImpl.cpp

```diff
@@ -998,10 +998,22 @@ ReturnCode_t DataWriterImpl::set_qos(
     {
         return ReturnCode_t::RETCODE_IMMUTABLE_POLICY;
     }
-    set_qos(qos_, qos_to_set, !enabled);
+
+    set_qos(qos_, qos_to_set, enabled);
```

Comment:
I don't like this change. This will not update immutable policies on a not-enabled DataWriter